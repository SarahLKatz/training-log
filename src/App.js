import React, { Component } from 'react';
// import SingleRun from './SingleRun';
import convert from 'convert-units';
import logo from './logo.svg';
import './App.css';
import styled, { css } from 'react-emotion';

// const stravaQueryHeaders = process.env.STRAVA_QUERY_HEADERS;

const sampleRun = {
  "id": 1394237509,
  "resource_state": 2,
  "external_id": "garmin_push_2481552282",
  "upload_id": 1504424724,
  "athlete": {
      "id": 20198690,
      "resource_state": 1
  },
  "name": "3.0 miles EC",
  "distance": 4854.3,
  "moving_time": 2260,
  "elapsed_time": 2260,
  "total_elevation_gain": 37.9,
  "type": "Run",
  "start_date": "2018-02-06T12:25:42Z",
  "start_date_local": "2018-02-06T07:25:42Z",
  "timezone": "(GMT-05:00) America/New_York",
  "utc_offset": -18000,
  "start_latlng": [
      40.72,
      -73.77
  ],
  "end_latlng": [
      40.72,
      -73.77
  ],
  "location_city": null,
  "location_state": null,
  "location_country": "United States",
  "start_latitude": 40.72,
  "start_longitude": -73.77,
  "achievement_count": 0,
  "kudos_count": 3,
  "comment_count": 0,
  "athlete_count": 1,
  "photo_count": 0,
  "map": {
      "id": "a1394237509",
      "summary_polyline": "_qpwFfogaM{CmCsED{Ha^_ZfS`H|WuLjFjLmGoFsMw@oH~[gTjB~LlBjBr@dKnGxL",
      "resource_state": 2
  },
  "trainer": false,
  "commute": false,
  "manual": false,
  "private": false,
  "flagged": false,
  "gear_id": null,
  "from_accepted_tag": false,
  "average_speed": 2.148,
  "max_speed": 2.9,
  "has_heartrate": false,
  "elev_high": 50,
  "elev_low": 22.2,
  "pr_count": 0,
  "total_photo_count": 0,
  "has_kudoed": false,
  "workout_type": null
}

class App extends Component {
  constructor() {
    super()
    this.state = {
      runs: {
        week1: {
          monday: 'OFF',
          tuesday: {
            planned: 3.0,
            type: 'EC',
            completed: convert(sampleRun.distance).from('m').to('mi').toFixed(2),
            status() {
              return (this.completed >= this.planned) ? allGood : fail;
            }
          },
          wednesday: 'OFF',
          thursday: {
            planned: 3.5,
            type: 'HM Tempo',
            completed: null,
            status() {
              return this.completed >= this.planned
            }
          },
          friday: 'OFF',
          saturday: 'OFF',
          sunday: {
            planned: 7.0,
            type: 'Easy/LR',
            completed: null,
            status() {
              return this.completed >= this.planned
            }
          }
        },
      }
    }
  }

  componentDidMount(){
    // axios.get('https://www.strava.com/api/v3/athlete/activities', { headers: {'Authorization': stravaQueryHeaders} })
    // .then(res => res.data)
    // .then(data => console.log(data))
  }

  render() {
    let weeks = Object.keys(this.state.runs)
    // let completed = convert(sampleRun.distance).from('m').to('mi').toFixed(2);
    // let status = (completed >= 3.0) ? allGood : fail;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Sarah's Training App</h1>
        </header>
        <table>
          <THead>
            <TR className={header}>
              <TD>Monday</TD>
              <TD>Tuesday</TD>
              <TD>Wednesday</TD>
              <TD>Thursday</TD>
              <TD>Friday</TD>
              <TD>Saturday</TD>
              <TD>Sunday</TD>
            </TR>
          </THead>
          <tbody>
            {
              weeks.map(week => {
                let {monday, tuesday, wednesday,thursday,friday,saturday,sunday} = this.state.runs[week];
                return (
                  <TR key={week}>
                    <TD>
                      {
                        typeof monday === 'string' ? 
                        monday :
                        <div>
                          <p>Planned: {monday.planned} {monday.type}</p>
                          {monday.completed && <p className={monday.status()}>Completed: {monday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof tuesday === 'string' ? 
                        tuesday :
                        <div>
                          <p>Planned: {tuesday.planned} {tuesday.type}</p>
                          { tuesday.completed && <p className={tuesday.status()}>Completed: {tuesday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof wednesday === 'string' ? 
                        wednesday :
                        <div>
                          <p>Planned: {wednesday.planned} {wednesday.type}</p>
                          { wednesday.completed && <p className={wednesday.status()}>Completed: {wednesday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof thursday === 'string' ? 
                        thursday :
                        <div>
                          <p>Planned: {thursday.planned} {thursday.type}</p>
                          { thursday.completed && <p className={thursday.status()}>Completed: {thursday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof friday === 'string' ? 
                        friday :
                        <div>
                          <p>Planned: {friday.planned} {friday.type}</p>
                          { friday.completed && <p className={friday.status()}>Completed: {friday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof saturday === 'string' ? 
                        saturday :
                        <div>
                          <p>Planned: {saturday.planned} {saturday.type}</p>
                          { saturday.completed && <p className={saturday.status()}>Completed: {saturday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                    <TD>
                      {
                        typeof sunday === 'string' ? 
                        sunday :
                        <div>
                          <p>Planned: {sunday.planned} {sunday.type}</p>
                          { sunday.completed && <p className={sunday.status()}>Completed: {sunday.completed} miles</p> }
                        </div>
                      }
                    </TD>
                  </TR>
                )
              })
            }
          </tbody>
        </table>
      </div>
    );
  }
}



const THead = styled('thead')`
  border: 3px solid black;
`

const header = css`
  color: rebeccapurple;
  font-weight: bold;
`

const TR = styled('tr')`
  border: 1px solid black;
`

const TD = styled('td')`
  border: 1px solid black;
`

const allGood = css`
  color: darkgreen;
  font-weight: bold;
`

const fail = css`
  color: red;
`

export default App;
