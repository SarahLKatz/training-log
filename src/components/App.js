import React, { Component } from 'react';
import Header from './Header';
import Training from './Training';
import { logInStrava } from '../utils/strava.js';

class App extends Component {
  constructor() {
    super()
    this.state = {
      userId: '',
      name: 'Runner'
    }
  }

  connectWithStrava = (e) => {
    e.preventDefault()
    logInStrava();
    this.setState({userId: '1234', name: 'Sarah'})
  }

  render() {
    return (
      <Header name={this.state.name}>
      {
        this.state.userId ?
        <Training />
        :
        <button onClick={this.connectWithStrava}>Connect With Strava</button>
      }
      </Header>
    )
  }
}

export default App;
