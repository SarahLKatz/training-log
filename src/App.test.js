import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

describe('main page' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('displays as expected', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot()
  });
});
