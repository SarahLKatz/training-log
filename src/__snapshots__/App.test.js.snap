// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page displays as expected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Sarah's Training App
    </h1>
  </header>
  <table>
    <thead>
      <tr
        className="css-dkmjt3"
      >
        <td
          className="css-13jbay9"
        >
          Monday
        </td>
        <td
          className="css-13jbay9"
        >
          Tuesday
        </td>
        <td
          className="css-13jbay9"
        >
          Wednesday
        </td>
        <td
          className="css-13jbay9"
        >
          Thursday
        </td>
        <td
          className="css-13jbay9"
        >
          Friday
        </td>
        <td
          className="css-13jbay9"
        >
          Saturday
        </td>
        <td
          className="css-13jbay9"
        >
          Sunday
        </td>
      </tr>
    </thead>
    <tbody>
      <tr
        className="css-tswxj3"
      >
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-5-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-6-2018
            </p>
            <p>
              Planned: 
              3
               
              EC
            </p>
            <p
              className="css-yylhj"
            >
              Completed: 
              3.02
               miles
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-7-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-8-2018
            </p>
            <p
              className="css-qafjh4"
            >
              Warm Up: 
              1
               Easy/LR
            </p>
            <p>
              Planned: 
              2
               
              HM Tempo
            </p>
            <p
              className="css-qafjh4"
            >
              Cool Down: 
              0.5
               Easy/LR
            </p>
            <button
              name="week2,thursday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-9-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-10-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-11-2018
            </p>
            <p>
              Planned: 
              7
               
              Easy/LR
            </p>
            <button
              name="week2,sunday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
      </tr>
      <tr
        className="css-tswxj3"
      >
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-12-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-13-2018
            </p>
            <p
              className="css-qafjh4"
            >
              Warm Up: 
              1
               Easy/LR
            </p>
            <p>
              Planned: 
              1
               
              10K
            </p>
            <p
              className="css-qafjh4"
            >
              Cool Down: 
              1
               Easy/LR
            </p>
            <button
              name="week3,tuesday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-14-2018
            </p>
            <p>
              Planned: 
              3.3
               
              Easy/LR
            </p>
            <button
              name="week3,wednesday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-15-2018
            </p>
            <p
              className="css-qafjh4"
            >
              Warm Up: 
              0.5
               Easy/LR
            </p>
            <p>
              Planned: 
              3
               
              HM Tempo
            </p>
            <p
              className="css-qafjh4"
            >
              Cool Down: 
              0.5
               Easy/LR
            </p>
            <button
              name="week3,thursday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-16-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-17-2018
            </p>
            <p>
               
              OFF
               
            </p>
          </div>
        </td>
        <td
          className="css-13jbay9"
        >
          <div>
            <p
              className="css-ddwizh"
            >
              2-18-2018
            </p>
            <p>
              Planned: 
              10
               
              Easy/LR
            </p>
            <button
              name="week3,sunday"
              onClick={[Function]}
            >
              Import Run(s)
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
